tests/interfaces/test_gen_object_detector_blackbox_sal.py

```diff
@@ -0,0 +1,246 @@
+    """
+    Test that the reference detection inputs must all have the same length.
+    """
+    m_impl = mock.Mock(spec=GenerateObjectDetectorBlackboxSaliency)
+    m_detector = mock.Mock(spec=DetectImageObjects)
+
+    test_image = np.ones((64, 64))
+
+    # Mismatched number of bboxes and scores, without objectness
+    test_bboxes = np.ones((4, 4), dtype=float)
+    test_scores = np.ones((5, 3), dtype=float)
+    with pytest.raises(
+        ValueError,
+        match=r"^Number of input bounding boxes and scores do not match: "
+              r"\(bboxes\) 4 != 5 \(scores\)$"
+    ):
+        GenerateObjectDetectorBlackboxSaliency.generate(
+            m_impl,
+            test_image,
+            test_bboxes,
+            test_scores,
+            m_detector
+        )
+
+    # Mismatched number of bboxes and scores, with objectness
+    test_bboxes = np.ones((5, 4), dtype=float)
+    test_scores = np.ones((4, 3), dtype=float)
+    test_objectness = np.ones((5,), dtype=float)
+    with pytest.raises(
+        ValueError,
+        match=r"^Number of input bounding boxes, scores, and objectness "
+              r"scores do not match: \(bboxes\) 5 != 4 \(scores\) and/or "
+              r"\(bboxes\) 5 != 5 \(objectness\)$"
+    ):
+        GenerateObjectDetectorBlackboxSaliency.generate(
+            m_impl,
+            test_image,
+            test_bboxes,
+            test_scores,
+            m_detector,
+            test_objectness
+        )
+
+    # Different number of objectness scores
+    test_bboxes = np.ones((5, 4), dtype=float)
+    test_scores = np.ones((5, 3), dtype=float)
+    test_objectness = np.ones((4,), dtype=float)
+    with pytest.raises(
+        ValueError,
+        match=r"^Number of input bounding boxes, scores, and objectness "
+              r"scores do not match: \(bboxes\) 5 != 5 \(scores\) and/or "
+              r"\(bboxes\) 5 != 4 \(objectness\)$"
+    ):
+        GenerateObjectDetectorBlackboxSaliency.generate(
+            m_impl,
+            test_image,
+            test_bboxes,
+            test_scores,
+            m_detector,
+            test_objectness
+        )
+
+
+def test_generate_checks_bboxes_width() -> None:
+    """
+    Test that the input bounding boxes must have a width of 4.
+    """
+    m_impl = mock.Mock(spec=GenerateObjectDetectorBlackboxSaliency)
+
+    m_detector = mock.Mock(spec=DetectImageObjects)
+
+    test_bboxes = np.ones((2, 3), dtype=float)
+    test_scores = np.ones((2, 3), dtype=float)
+
+    test_image = np.ones((256, 256), dtype=np.uint8)
+    with pytest.raises(
+        ValueError,
+        match=r"^Input bounding boxes matrix has width of 3, should have width of 4$"
+    ):
+        GenerateObjectDetectorBlackboxSaliency.generate(
+            m_impl,
+            test_image,
+            test_bboxes,
+            test_scores,
+            m_detector,
+        )
+
+
+def test_generate_checks_output_mismatch() -> None:
```

Comment:
Aligning with the interface comment, we could add a congruent test that makes sure we check the quantity of heatmaps output against the number of detections input.